From 7f46adf250a2076890f67110fd4c8d5fe059fc17 Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Fri, 3 Jul 2020 10:22:57 +0800
Subject: [PATCH] policy/modules/services/zebra: make zebra_t domain MLS
 trusted for reading from files up to its clearance

* Allow zebra_t to search /run/systemd/journal
* Allow zebra_t to read nsfs_t

Fixes:
avc:  denied  { search } for  pid=242 comm="zebra" name="journal"
dev="tmpfs" ino=10990 scontext=system_u:system_r:zebra_t:s0-s15:c0.c1023
tcontext=system_u:object_r:syslogd_runtime_t:s15:c0.c1023 tclass=dir
permissive=0

avc:  denied  { read } for  pid=242 comm="zebra" dev="nsfs"
ino=4026531992 scontext=system_u:system_r:zebra_t:s0-s15:c0.c1023
tcontext=system_u:object_r:nsfs_t:s0 tclass=file permissive=0

Upstream-Status: Pending

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/services/zebra.te | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/policy/modules/services/zebra.te b/policy/modules/services/zebra.te
index f20b45a28..1be7f96a4 100644
--- a/policy/modules/services/zebra.te
+++ b/policy/modules/services/zebra.te
@@ -104,6 +104,7 @@ files_read_etc_runtime_files(zebra_t)
 
 fs_getattr_all_fs(zebra_t)
 fs_search_auto_mountpoints(zebra_t)
+fs_read_nsfs_files(zebra_t)
 
 term_list_ptys(zebra_t)
 
@@ -116,6 +117,8 @@ sysnet_read_config(zebra_t)
 userdom_dontaudit_use_unpriv_user_fds(zebra_t)
 userdom_dontaudit_search_user_home_dirs(zebra_t)
 
+mls_file_read_to_clearance(zebra_t)
+
 tunable_policy(`allow_zebra_write_config',`
 	manage_files_pattern(zebra_t, zebra_conf_t, zebra_conf_t)
 ')
-- 
2.17.1

